### ISSUE TITLE
Link reference definitions with \r\n line endings are not parsed correctly

### ISSUE THREAD

**Issue by Minecrell** [vsch/flexmark-java#77]
If you have a document like the following:

```markdown
[flexmark-java] [flexmark-java issues]

[flexmark-java]: https://github.com/vsch/flexmark-java
[flexmark-java issues]: https://github.com/vsch/flexmark-java/issues
```

It will correctly parse when it has only `\n` as line separator:

```html
<p><a href="https://github.com/vsch/flexmark-java">flexmark-java</a> <a href="https://github.com/vsch/flexmark-java/issues">flexmark-java issues</a></p>
```

However, if the line separator is `\r\n` and you have at least two consecutive link reference definitions then it is not parsed correctly:

```html
<p>[flexmark-java] [flexmark-java issues]</p>
<p>[flexmark-java]: https://github.com/vsch/flexmark-java
[flexmark-java issues]: https://github.com/vsch/flexmark-java/issues</p>
```

Can be reproduced using the following Java code:

```java
import com.vladsch.flexmark.ast.Node;
import com.vladsch.flexmark.html.HtmlRenderer;
import com.vladsch.flexmark.parser.Parser;

public class FlexmarkTest {

    public static void main(String[] args) {
        Parser parser = Parser.builder().build();
        HtmlRenderer renderer = HtmlRenderer.builder().indentSize(4).build();

        String input = "[flexmark-java] [flexmark-java issues]\n"
                + "\n"
                + "[flexmark-java]: https://github.com/vsch/flexmark-java\n"
                + "[flexmark-java issues]: https://github.com/vsch/flexmark-java/issues\n";

        // With \n it is parsed correctly
        Node node = parser.parse(input);
        System.out.println(renderer.render(node));

        // With \r\n it is not parsed correctly
        node = parser.parse(input.replace("\n", "\r\n"));
        System.out.println(renderer.render(node));
    }

}
```

**Comment by vsch** [comment]
@Minecrell, thank you for reporting this. It is was caused by many instances of `\n` in regular expressions used for parsing instead of `\r\n|\r|\n` as needed since the original regex was from commonmark-java which normalized EOL before parsing and flexmark-java does it in rendering to keep the AST source based.

This bug is a duplicate of #75 and was fixed in version 0.18.1, current version is 0.18.2 and has a few more bug fixes.

I tried your sample and the output is:

```html
<p><a href="https://github.com/vsch/flexmark-java">flexmark-java</a> <a href="https://github.com/vsch/flexmark-java/issues">flexmark-java issues</a></p>

<p><a href="https://github.com/vsch/flexmark-java">flexmark-java</a> <a href="https://github.com/vsch/flexmark-java/issues">flexmark-java issues</a></p>
```

confirming that this is fixed in the latest release.
